pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -163,28 +163,25 @@ func (p *DefaultProvider) createAMIOptions(ctx context.Context, nodeClass *v1bet
 	if err != nil {
 		return nil, err
 	}
+	// Since we are using security group status, A node is launched and immediately drift due to
+	// consistency delay of when the security group status is updated
+	// TODO @aengeda: add observation check to validate that the status is updated before a node is launched
```

Comment:
This reads a little confusing and vague to me. WDYT:
```suggestion
	// Relying on the status rather than an API call means that Karpenter is subject to a race
	// condition where EC2NodeClass spec changes haven't propagated to the status once a node 
	// has launched. 
	// If a user changes their EC2NodeClass and shortly after Karpenter launches a node,
	// in the worst case, the node could be drifted and re-created.
	// TODO @aengeda: add status generation fields to gate node creation until the status is updated from a spec change
```